server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -272,6 +272,37 @@ public class IndicesService extends AbstractLifecycleComponent
         Property.Final
     );
 
+    /**
+     * This setting is used to set the refresh interval when the {@code index.refresh_interval} index setting is not
+     * provided during index creation or when the exiting {@code index.refresh_interval} index setting is set as null.
+     * This comes handy when the user wants to set a default refresh interval across all indexes created in a cluster
+     * which is different from 1s and also at the same time have searchIdle feature supported. The setting can only be
+     * as low as the {@code cluster.minimum.index.refresh_interval}.
+     */
+    public static final Setting<TimeValue> CLUSTER_DEFAULT_INDEX_REFRESH_INTERVAL_SETTING = Setting.timeSetting(
+        "cluster.default.index.refresh_interval",
+        IndexSettings.DEFAULT_REFRESH_INTERVAL,
+        IndexSettings.MINIMUM_REFRESH_INTERVAL,
+        new ClusterDefaultRefreshIntervalValidator(),
+        Property.NodeScope,
+        Property.Dynamic
+    );
+
+    /**
+     * This setting is used to set the minimum refresh interval applicable for all indexes in a cluster. The
+     * {@code cluster.default.index.refresh_interval} setting value needs to be higher than this setting's value. Index
+     * creation will fail if the index setting {@code index.refresh_interval} is supplied with a value greater than the
+     * cluster minimum refresh interval.
+     */
+    public static final Setting<TimeValue> CLUSTER_MINIMUM_INDEX_REFRESH_INTERVAL_SETTING = Setting.timeSetting(
+        "cluster.minimum.index.refresh_interval",
```

Comment:
what would happen in the following case:
1. cluster minimum is 5 sec, cluster default is 20 sec
2. index created with explicit refresh interval of 8secs
3. what happens when the cluster minimum setting is changed to 10 secs. do we keep the index created with 8 sec refresh interval as is?

also, we should add a test around this. looking at the existing tests we did not cover this case